pkg/querier/blocks_store_queryable.go

```diff
@@ -675,7 +693,11 @@ func canBlockWithCompactorShardIndexContainQueryShard(queryShardIndex, queryShar
 // Errors while creating storepb.SeriesRequest, context cancellation, and unprocessable
 // requests to the store-gateways (e.g., if a chunk or series limit is hit) are
 // considered serious errors. All other errors are not returned, but they give rise to fetch retrials.
-func (q *blocksStoreQuerier) fetchSeriesFromStores(ctx context.Context, sp *storage.SelectHints, clients map[BlocksStoreClient][]ulid.ULID, minT int64, maxT int64, convertedMatchers []storepb.LabelMatcher) ([]storage.SeriesSet, []ulid.ULID, storage.Warnings, error) {
+//
+// In case of a successful run, fetchSeriesFromStores returns a streamCloser function if it was a streaming
+// call for series+chunks. If you are ending the execution of the query later without iterating through all the series
+// and consuming the chunks, the streamCloser MUST be called to avoid leaking goroutines and gRPC connections.
```

Comment:
Is this comment still correct? Looks like we've moved responsibility for starting streaming out of this method.